packages/terra-application/src/notification-dialog/NotificationDialog.jsx

```diff
@@ -0,0 +1,289 @@
+   * the start of the notification dialog body. Don’t repeat the title verbatim.
+   */
+  startMessage: PropTypes.string,
+  /**
+   * The text to provide more detail or defined terminology to be displayed at
+   * the end of the notification dialog body. Don’t repeat the title verbatim.
+   */
+  endMessage: PropTypes.string,
+  /**
+   * The content to be inserted after `startMessage` and/or before `endMessage`
+   * to provide more details to the user in the dialog body. Don’t repeat the
+   * title verbatim.
+   */
+  content: PropTypes.node,
+  /**
+   * The button text and onclick values of the accept button.
+   */
+  acceptAction: PropTypes.shape({
+    text: PropTypes.string,
+    onClick: PropTypes.func,
+  }),
+  /**
+   * The button text and onclick values of the reject button.
+   */
+  rejectAction: PropTypes.shape({
+    text: PropTypes.string,
+    onClick: PropTypes.func,
+  }),
+  /**
+   * Determines the order of notification action buttons.
+   */
+  buttonOrder: PropTypes.oneOf([
+    'acceptFirst',
+    'rejectFirst',
+  ]),
+  /**
+   * Determines whether acceptAction, rejectAction or neither is emphasized.
+   */
+  emphasizedAction: PropTypes.oneOf([
+    'none',
+    'accept',
+    'reject',
+  ]),
+  /**
+   * The pieces to populate a notification-dialog when `variant="custom"`.
+   */
+  custom: PropTypes.shape({
+    /**
+     * The keyword used to represent & emphasis the intention of dialog message
+     * that is being shown to the user.
+     */
+    signalWord: PropTypes.string,
+    /**
+     * The class name used to set the icon as the background image to be used as
+     * the icon in the notification-dialog.
+     */
+    iconClassName: PropTypes.string,
+  }),
+};
+
+const defaultProps = {
+  buttonOrder: 'acceptFirst',
+  emphasizedAction: 'none',
+  custom: {},
+};
+
+const actionSection = (acceptAction, rejectAction, buttonOrder, emphasizedAction) => {
+  if (!acceptAction && !rejectAction) {
+    return null;
+  }
+
+  const actionButtons = [];
+
+  if (acceptAction) {
+    actionButtons.push((
+      <Button
+        key="accept"
+        variant={emphasizedAction === 'accept' ? 'emphasis' : 'neutral'}
+        text={acceptAction.text}
+        onClick={acceptAction.onClick}
+      />
+    ));
+  }
+
+  if (rejectAction) {
+    actionButtons.push((
+      <Button
+        key="reject"
+        variant={emphasizedAction === 'reject' ? 'emphasis' : 'neutral'}
+        text={rejectAction.text}
+        onClick={rejectAction.onClick}
+      />
+    ));
+  }
+
+  return (
+    <div className={cx('actions')}>
+      {buttonOrder === 'acceptFirst' ? actionButtons : actionButtons.reverse()}
+    </div>
+  );
+};
+
+/**
+ * Custom hook to manage Tab key presses within a given container.
+ * @param {Object} containerRef React ref object containing current value of
+ * the element to trap focus within.
+ */
+const useFocusTrap = (containerRef) => {
```

Comment:
I like this. Good work. 👍

Reply:
Using inert everywhere allows us to keep this pretty simple. Should be able to put it in the new modals as well, I can externalize the hook when that is the case.